src/Helpers.hs

```diff
@@ -0,0 +1,69 @@
+module Helpers
+    ( getFilePath
+    , inc
+    , indentedOutput
+    , loadEntries
+    , nextRowNum
+    , parseEntries
+    , parseEntry
+    , printEntries
+    , showOutput
+    ) where
+
+
+import           Data.List        (lines, unlines)
+import           Entry
+import           System.Directory (getAppUserDataDirectory)
+import           System.IO        (readFile)
+
+
+getFilePath :: IO FilePath
+getFilePath = do
+    path <- getAppUserDataDirectory "timetrack"
+    return $ path ++ "/timetrack.txt"
+
+
+inc :: Integer -> Integer
+inc =
+    (+1)
+
+
+indentedOutput :: String -> String
+indentedOutput str =
+    "   ├── " ++ str
+
+
+nextRowNum :: [Entry] -> Integer
+nextRowNum =
+    inc . fromIntegral . length
+
+
+loadEntries :: IO [String]
+loadEntries = do
+    path <- getFilePath
+    contents <- readFile path
+    return $ lines contents
+
+
+parseEntries :: [String] -> [Entry]
+parseEntries =
+    zipWith parseEntry [1..]
+
+
+parseEntry :: Integer -> String -> Entry
+parseEntry n line =
+    let
+        (date, rest) = splitAt 10 line
+        message = drop 1 rest
+    in
+        Entry { index = n, date = date, message = message }
+
+
+printEntries :: [Entry] -> IO ()
+printEntries =
+    putStrLn . unlines . fmap show
```

Comment:
Not huge on all these `entries`-specific helpers being in here

Reply:
I'd probably put them in `src/Entry.hs` - that's a good place for any function that returns an `Entry` (something like a class method on the "Entry" class from OO land) or takes an `Entry` as its first argument (something like an instance method). I'll often just name the function `parse` or `print` and plan on importing them qualified:

```
import qualified Entry

parseEntries :: [String] -> [Entry]
parseEntries = zipWith Entry.parse [1..]
```

Reply:
Yes, that does seem to group things like this better